src/devtools/client/webconsole/components/App.js

```diff
@@ -3,110 +3,35 @@
  * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
 "use strict";
 
-const Services = require("Services");
-const { Component, createFactory } = require("react");
+const React = require("react");
 const PropTypes = require("prop-types");
-const dom = require("react-dom-factories");
 const { connect } = require("devtools/client/shared/redux/visibility-handler-connect");
-
-const actions = require("devtools/client/webconsole/actions/index");
 const { FILTERBAR_DISPLAY_MODES } = require("devtools/client/webconsole/constants");
 
 // We directly require Components that we know are going to be used right away
-const ConsoleOutput = createFactory(
-  require("devtools/client/webconsole/components/Output/ConsoleOutput")
-);
-const FilterBar = createFactory(
-  require("devtools/client/webconsole/components/FilterBar/FilterBar")
-);
-const ReverseSearchInput = createFactory(
-  require("devtools/client/webconsole/components/Input/ReverseSearchInput")
-);
-const JSTerm = createFactory(require("devtools/client/webconsole/components/Input/JSTerm"));
-
-const EagerEvaluation = createFactory(
-  require("devtools/client/webconsole/components/Input/EagerEvaluation")
-);
-
-const GridElementWidthResizer = createFactory(
-  require("devtools/client/shared/components/splitter/GridElementWidthResizer")
-);
-
-const l10n = require("devtools/client/webconsole/utils/l10n");
-
-const { div } = dom;
-const isMacOS = Services.appinfo.OS === "Darwin";
+const ConsoleOutput = require("devtools/client/webconsole/components/Output/ConsoleOutput");
+const FilterBar = require("devtools/client/webconsole/components/FilterBar/FilterBar");
+const JSTerm = require("devtools/client/webconsole/components/Input/JSTerm");
 
 require("./App.css");
 
 /**
  * Console root Application component.
  */
-class App extends Component {
+class App extends React.Component {
   static get propTypes() {
     return {
       dispatch: PropTypes.func.isRequired,
-      notifications: PropTypes.object,
-      autocomplete: PropTypes.bool,
-      currentReverseSearchEntry: PropTypes.string,
-      reverseSearchInputVisible: PropTypes.bool,
-      reverseSearchInitialValue: PropTypes.string,
-      editorMode: PropTypes.bool,
-      editorWidth: PropTypes.number,
-      sidebarVisible: PropTypes.bool.isRequired,
-      eagerEvaluationEnabled: PropTypes.bool.isRequired,
       filterBarDisplayMode: PropTypes.oneOf([...Object.values(FILTERBAR_DISPLAY_MODES)]).isRequired,
     };
   }
 
-  constructor(props) {
-    super(props);
-
-    this.onClick = this.onClick.bind(this);
-    this.onKeyDown = this.onKeyDown.bind(this);
-    this.onBlur = this.onBlur.bind(this);
-  }
-
   componentDidMount() {
     window.addEventListener("blur", this.onBlur);
```

Comment:
Can delete this - there's no more `onBlur()`

Reply:
definitely